<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>bam2dm &mdash; DMtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            DMtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Getting started with dmtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmtools.html">dmtools: command line tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmplot.html">dmplot: plotting from the command line</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmdiff.html">dmdiff: differential dna methylation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pydmtools.html">Python API: pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">DMtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">bam2dm</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/bam2dm.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="bam2dm">
<h1>bam2dm<a class="headerlink" href="#bam2dm" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage" id="id2">Usage</a></p></li>
<li><p><a class="reference internal" href="#paramaters" id="id3">Paramaters</a></p></li>
<li><p><a class="reference internal" href="#output-files" id="id4">Output files</a></p></li>
<li><p><a class="reference internal" href="#output-file-format" id="id5">Output file format</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>Calculate DNA methylation level from alignment files, you can obtained single-base cytosine DNA
methylation results, and the chromosome region DNA methylation levels files.</p>
<p>To calculate the methylation density, we first count the total number of C/T nucleotides that
overlap with each cytosine site on the plus strand and the number of G/A nucleotides on
the minus strand.
Those cytosines, which are used for further statistical analysis,
should meet the criterion that their depth (C plus T) should be more than some predefined
threshold (by default, 5) to reduce the influence of sequencing errors in the cytosine site.</p>
<p>We calculate the methylation level of postion <img class="math" src="../_images/math/5aa339d4daf45a810dda332e3c80a0698e526e04.png" alt="i"/> (<img class="math" src="../_images/math/59ab811af3258655a8616339386e1419c632b592.png" alt="ML(i)"/>) by the following equation:</p>
<div class="math">
<p><img src="../_images/math/e4003620d01a52535637b0a9e7509039bb142a20.png" alt="ML(i) = \dfrac{C_{i}}{C_{i} + T_{i}}"/></p>
</div></section>
<section id="usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage</a><a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>An example usage is:
  with bam file:
    dmtools bam2dm <span class="o">[</span>options<span class="o">]</span> -g genome.fa -b alignment.sort.bam -m output.methrario.dm
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The sam or bam file MUST sorted by <cite>samtools sort</cite>.</p>
</div>
</section>
<section id="paramaters">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Paramaters</a><a class="headerlink" href="#paramaters" title="Permalink to this heading"></a></h2>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head" colspan="3"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>-m/--methratio</p></td>
<td><p>[MethFileNamePrefix]  Predix of methratio output file (dm)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--genome/-g</p></td>
<td><p>Name of the genome mapped against, MUST build index first <span class="xref std std-doc">bdindex</span></p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-i/--input</p></td>
<td><p>Sam format file, sorted by samtools sort.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b/--binput</p></td>
<td><p>Bam format file, sorted by samtools sort.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-Q [int]</p></td>
<td><p>caculate the methratio while read QulityScore &gt;= Q. default:20</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-n [float]</p></td>
<td><p>Number of mismatches, default 0.06 percentage of read length. [0-1]</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-c|--coverage</p></td>
<td><p>&gt;= &lt;INT&gt; coverage. default:4</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-nC</p></td>
<td><p>&gt;= &lt;INT&gt; Cs per region. default:1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-r/--remove_dup</p></td>
<td><p>REMOVE_DUP, default:true</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--zl</p></td>
<td><p>The maximum number of zoom levels. [1-10], default: 2</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-as [0/1]</p></td>
<td><p>If print calculated alignment reads in sam/bam file. default:0</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--mrtxt</p></td>
<td><p>also print prefix.methratio.txt file</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--help/-h</p></td>
<td><p>Print help</p></td>
</tr>
</tbody>
</table>
</section>
<section id="output-files">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Output files</a><a class="headerlink" href="#output-files" title="Permalink to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. prefix.methratio.dm <span class="o">(</span>binary file with index, view and processed with dmtools<span class="o">)</span>
</pre></div>
</div>
</section>
<section id="output-file-format">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Output file format</a><a class="headerlink" href="#output-file-format" title="Permalink to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. methratio.dm <span class="o">(</span>binary file, view and processed with dmtools<span class="o">)</span>
    Chromosome Loci <span class="o">(</span>end<span class="o">)</span> methlevel CT_count Strand Context
    <span class="c1"># ex. dmtools view -i test.mr.dm -r Chr1:61-61</span>
    <span class="c1"># ex. Chr1    61      0.286364  11  +       CHH</span>
    <span class="c1"># CT_count     The number of coverage in this base pair.</span>
    <span class="c1"># BigWig for genome browser, suggest with end col.</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>