<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmtools bodystats &mdash; dmtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            dmtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="dmtools.html">DMtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">DiffMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pydmtools.html">pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">dmtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">dmtools bodystats</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/bodystats.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmtools-bodystats">
<h1>dmtools bodystats<a class="headerlink" href="#dmtools-bodystats" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage-and-output" id="id2">Usage and output</a></p></li>
<li><p><a class="reference internal" href="#parameters" id="id3">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>DMtools provides a <code class="docutils literal notranslate"><span class="pre">bodystats</span></code> command that can be used to calculate DNA methylation levels in user-specified regions of interest and their upstream and downstream regions. This feature is useful for researchers who may be interested in the methylation levels of specific genes, transposable elements (TEs), or other specified regions, as well as their upstream and downstream regions.</p>
<p>While the &quot;regionstats&quot; command can only calculate DNA methylation levels in specific regions, &quot;bodystats&quot; can calculate methylation levels in both the specified regions and their upstream and downstream regions, by combining with the &quot;--regionextend&quot; parameter.</p>
<p>By using the dmtools bodystats command, researchers can quickly and easily calculate the methylation levels in each region of interest as well as its upstream and downstream regions, and conduct further analysis and comparison.</p>
<p>To calculate the methylation density level in a given genomic region,
only cytosines with coverage greater than the preset threshold are used.
The DNA methylation level in a genomic region is defined as the total number of sequenced Cs over the
total number of sequenced Cs and Ts at all cytosine positions across the region,
and the equation is as follows:</p>
<div class="math">
<p><img src="../_images/math/5df321b8f1e45c7229dcb040c5c7a54e1ddf7be6.png" alt="ML(x)_{method} = \begin{cases}
\dfrac{\sum_{i=1}^{N} C}{\sum_{i=1}^{N} (C + T)}, &amp; \text{if method = weighted} \\
\dfrac{\sum_{i=1}^{N} ML(x_{i})}{N}, &amp; \text{if method = mean} \\
\end{cases}"/></p>
</div><p>where N is the total number of cytosine sites whose coverage is more than the predefined threshold in the genomic region.</p>
<p>Calculate DNA methylation level of gene body, upstream and downstream:</p>
</section>
<section id="usage-and-output">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage and output</a><a class="headerlink" href="#usage-and-output" title="Permalink to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools bodystats -i sample1.methratio.dm --gtf gene.gtf -o gene.meth.txt
</pre></div>
</div>
<p>Or with bed file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools bodystats -i sample1.methratio.dm --bed gene.bed -o gene.meth.txt
</pre></div>
</div>
<p>Or just calculate DNA methylation level of same regions:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools bodystats -i sample1.methratio.dm -r chr1:1-2900<span class="p">;</span>chr2:1-200,+ <span class="se">\</span>
  -o gene.meth.txt
</pre></div>
</div>
<p>Please see 'dmtools bodystats' for more details.</p>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">-i</span></code> input DM file</p>
<p><code class="docutils literal notranslate"><span class="pre">--bed</span></code> bed file for view, format: chrom start end [strand].</p>
<p><code class="docutils literal notranslate"><span class="pre">--gtf</span></code> gtf file for view, format: chrom * * start end * strand * xx geneid.</p>
<p><code class="docutils literal notranslate"><span class="pre">--gff</span></code> gff file for view, format: chrom * * start end * strand * xx=geneid.</p>
<p><code class="docutils literal notranslate"><span class="pre">-o</span></code> output file [stdout]</p>
<p><code class="docutils literal notranslate"><span class="pre">-r</span></code> region for view, can be seperated by space. chr1:1-2900 chr2:1-200,+</p>
<p><code class="docutils literal notranslate"><span class="pre">--method</span></code> weighted/ mean</p>
<p><code class="docutils literal notranslate"><span class="pre">--regionextend</span></code> also calculate DNA methylation level of upstream and downstream N-bp window. default 2000.</p>
<p><code class="docutils literal notranslate"><span class="pre">--strand</span></code> [0/1/2/3] strand for show, 0 represent '+' positive strand, 1 '-' negative strand, 2 '.' all information, 3 calculate and print strand meth level seperately, only valid while -r para</p>
<p><code class="docutils literal notranslate"><span class="pre">--context</span></code> [0/1/2/3/4] context for show, 0 represent 'C/ALL' context, 1 'CG' context, 2 'CHG' context, 3 'CHH' context, 4 calculate and print strand meth level seperately, default: 4.</p>
<p><code class="docutils literal notranslate"><span class="pre">--printcoverage</span></code> [0/1] also print countC and coverage of body instead of methratio. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">--print2one</span></code> [int] print all the countC and coverage results of C/CG/CHG/CHH context methylation to same file, only valid when --printcoverage 1. 0 for no, 1 for yes. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">-h|--help</span></code></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>