<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmtools chromstats &mdash; DMtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="dmtools chrmeth" href="chrmeth.html" />
    <link rel="prev" title="dmtools profile" href="profile.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            DMtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Getting started with dmtools</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../dmtools.html">dmtools: command line tools</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bam2dm.html">Calculate DNA meth from bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-format.html">Format convertion</a></li>
<li class="toctree-l2"><a class="reference internal" href="ebsrate.html">Estimate bs conversion rate</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="dmtools-calme.html">Calculate DNA meth with dm</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="stats.html">Calculate DNA methylation for dm file</a></li>
<li class="toctree-l3"><a class="reference internal" href="regionstats.html">Calculate DNA methylation for regions</a></li>
<li class="toctree-l3"><a class="reference internal" href="bodystats.html">Calculate DNA methylation for gene body</a></li>
<li class="toctree-l3"><a class="reference internal" href="profile.html">Calculate methylation profile across genes</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Calculate DNA methylation acorss chromosome</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l4"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l4"><a class="reference internal" href="#parameters">Parameters</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="chrmeth.html">Calculate DNA methylation for chromosome</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-merge.html">Merge or overlap dm files</a></li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-manu.html">Manipulate dm format</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../dmplot.html">dmplot: plotting from the command line</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">dmdiff: differential dna methylation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pydmtools.html">Python API: pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">DMtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../dmtools.html">Getting started</a></li>
          <li class="breadcrumb-item"><a href="dmtools-calme.html">DMtools main modules</a></li>
      <li class="breadcrumb-item active">dmtools chromstats</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/chromstats.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmtools-chromstats">
<h1>dmtools chromstats<a class="headerlink" href="#dmtools-chromstats" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage" id="id2">Usage</a></p></li>
<li><p><a class="reference internal" href="#parameters" id="id3">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>DMtools can calculate and visualize cytosine methylation density at the chromosome level.
The bins represent a sliding window of N-kb with a step of S-kb.
N represent --chromstep paramater and S represent --stepmove paramater.</p>
<p>To calculate the methylation density level in a given genomic region,
only cytosines with coverage greater than the preset threshold are used.
The DNA methylation level in a genomic region is defined as the total number of sequenced Cs over the
total number of sequenced Cs and Ts at all cytosine positions across the region,
and the equation is as follows:</p>
<div class="math notranslate nohighlight">
\[\begin{split}ML(x)_{method} = \begin{cases}
\dfrac{\sum_{i=1}^{N} C}{\sum_{i=1}^{N} (C + T)}, &amp; \text{if method = weighted} \\
\dfrac{\sum_{i=1}^{N} ML(x_{i})}{N}, &amp; \text{if method = mean} \\
\end{cases}\end{split}\]</div>
<p>where N is the total number of cytosine sites whose coverage is more than the predefined threshold in the genomic region.</p>
<p>Calculate DNA methylation level across chromosome</p>
</section>
<section id="usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage</a><a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools chromstats -i sample1.methratio.dm -o chromosome.meth.txt <span class="se">\</span>
  --chromstep <span class="m">100000</span> --stepmove <span class="m">50000</span>
<span class="c1">## output file format `chromosome.meth.txt`</span>
<span class="c1">## chrom start end methlevel context strand</span>
<span class="c1">#chrM    0       16569   0.004313        C       .</span>
<span class="c1">#chrM    0       16569   0.005997        CG      .</span>
<span class="c1">#chrM    0       16569   0.003897        CHG     .</span>
<span class="c1">#chrM    0       16569   0.003890        CHH     .</span>
<span class="c1"># ... ...</span>
<span class="c1">#chr1    200000  300000  0.022242        C       .</span>
<span class="c1">#chr1    200000  300000  0.313409        CG      .</span>
<span class="c1">#chr1    200000  300000  0.003797        CHG     .</span>
<span class="c1">#chr1    200000  300000  0.005633        CHH     .</span>
</pre></div>
</div>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">-i</span></code> input dm file</p>
<p><code class="docutils literal notranslate"><span class="pre">-o</span></code> output file of calculated DNA methylation levels for all bins</p>
<p><code class="docutils literal notranslate"><span class="pre">--method</span></code> weighted/ mean/ min/ max/ cover/ dev methods for calculate DNA methylation in per bin</p>
<p><code class="docutils literal notranslate"><span class="pre">--chromstep</span></code> [int] step mean bin size for chromosome region, default: 100000</p>
<p><code class="docutils literal notranslate"><span class="pre">--stepmove</span></code> [int] step move, default: 50000, if no overlap, please define same as --chromstep</p>
<p><code class="docutils literal notranslate"><span class="pre">--fstrand</span></code> [0/1/2/3] strand for calculation, 0 represent '+' positive strand, 1 '-' negative strand, 2 '.' without strand information, 3 calculate and print strand meth level seperately. [2]</p>
<p><code class="docutils literal notranslate"><span class="pre">--context</span></code> [0/1/2/3/4] context for calculate, 0 represent 'C/ALL' context, 1 'CG' context, 2 'CHG' context, 3 'CHH' context, 4 calculate and print context meth level seperately. [4]</p>
<p><code class="docutils literal notranslate"><span class="pre">--printcoverage</span></code> [0/1] print countC and coverage instead of methratio. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">--print2one</span></code> [int] print all the countC and coverage results of C/CG/CHG/CHH context methylation to same file, only valid when --printcoverage 1. 0 for no, 1 for yes. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">-h|--help</span></code> usage</p>
<p>Please see 'dmtools chromstats' for more details.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="profile.html" class="btn btn-neutral float-left" title="dmtools profile" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="chrmeth.html" class="btn btn-neutral float-right" title="dmtools chrmeth" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>