<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmPlotHeatmap &mdash; DMtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="dmPlotProfile" href="dmplotprofile.html" />
    <link rel="prev" title="dmPlotCor" href="dmplotcor.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            DMtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Getting started with dmtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmtools.html">dmtools: command line tools</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../dmplot.html">dmplot: plotting from the command line</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="dmplotbasic.html">dmplotbasic</a></li>
<li class="toctree-l2"><a class="reference internal" href="dmplotcor.html">dmplotcor</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">dmplotheatmap</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l3"><a class="reference internal" href="#usage-and-example">Usage and example</a></li>
<li class="toctree-l3"><a class="reference internal" href="#parameters">Parameters</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="dmplotprofile.html">dmplotprofile</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">dmdiff: differential dna methylation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pydmtools.html">Python API: pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">DMtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../dmplot.html">Getting started</a></li>
      <li class="breadcrumb-item active">dmPlotHeatmap</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/dmplotheatmap.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmplotheatmap">
<h1>dmPlotHeatmap<a class="headerlink" href="#dmplotheatmap" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage-and-example" id="id2">Usage and example</a></p></li>
<li><p><a class="reference internal" href="#parameters" id="id3">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>Heatmap visualization for DNA methylation across genes or DMRs.
The input files can be generated by <a class="reference internal" href="profile.html"><span class="doc">dmtools profile</span></a> and <a class="reference internal" href="bodystats.html"><span class="doc">dmtools bodystats</span></a> command.</p>
</section>
<section id="usage-and-example">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage and example</a><a class="headerlink" href="#usage-and-example" title="Permalink to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools profile -i sample1.methratio.dm --bed H3K4me3.bdgene.bed -o H3K4me3.bdgene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">0</span>

$ python dmPlotHeatmap.py -m H3K4me3.bdgene.profile.cg -l <span class="nb">bg</span> <span class="se">\</span>
-o test0.pdf -z k43 -sl TSS -el TTS
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/plot-heatmap-0.png"><img alt="heatmap0" class="align-center" src="../_images/plot-heatmap-0.png" style="width: 200px; height: 380px;" /></a>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools profile -i sample1.methratio.dm --bed H3K4me3.bdgene.bed -o H3K4me3.bdgene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">1</span>

$ dmtools profile -i sample1.methratio.dm --bed H3K4me3.bdgene.bed -o H3K4me3.bdgene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">2</span>

$ python dmPlotHeatmap.py -m H3K4me3.bdgene.profile.tss.cg H3K4me3.bdgene.profile.tts.cg <span class="se">\</span>
    -l tss tts -o test.pdf --zMax <span class="m">0</span>.1 --colorMap vlag --centerlabel center -z bd
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/plot-heatmap-1.png"><img alt="heatmap0" class="align-center" src="../_images/plot-heatmap-1.png" style="width: 400px; height: 460px;" /></a>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ python dmPlotHeatmap.py -m H3K4me3.bdgene.profile.tss.cg H3K4me3.bdgene.profile.tts.cg <span class="se">\</span>
    H3K4me3.unbdgene.profile.tss.cg H3K4me3.unbdgene.profile.tts.cg <span class="se">\</span>
    -l <span class="nb">test</span> end -o test2.pdf --zMax <span class="m">0</span>.05 --centerlabel center <span class="se">\</span>
    --plotmatrix 2x2 --colorList white,red -z bd unbd
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/plot-heatmap-2.png"><img alt="heatmap0" class="align-center" src="../_images/plot-heatmap-2.png" style="width: 400px; height: 500px;" /></a>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools bodystats --gtf H3K4me3.bdgene.gtf -i ./test.methratio.dm <span class="se">\</span>
  -o H3K4me3.bdgene --strand <span class="m">3</span> --context <span class="m">4</span> --printcoverage <span class="m">1</span>

$ python dmPlotHeatmap.py -f H3K4me3.bdgene.bodym.cover.cg H3K4me3.bdgene.bodym.cover.cg <span class="se">\</span>
    H3K4me3.unbdgene.bodym.cover.cg H3K4me3.unbdgene.bodym.cover.cg <span class="se">\</span>
    -l <span class="nb">test</span> end -o test3.pdf --zMax <span class="m">0</span>.5 --centerlabel center <span class="se">\</span>
    --plotmatrix 2x2 -z bd unbd
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/plot-heatmap-3.png"><img alt="heatmap0" class="align-center" src="../_images/plot-heatmap-3.png" style="width: 400px; height: 500px;" /></a>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ python dmPlotHeatmap.py -m H3K4me3.bdgene.profile.tss.cg H3K4me3.bdgene.profile.tts.cg <span class="se">\</span>
    H3K4me3.bdgene.profile.tss.chg H3K4me3.bdgene.profile.tts.chg <span class="se">\</span>
    H3K4me3.bdgene.profile.tss.chh H3K4me3.bdgene.profile.tts.chh <span class="se">\</span>
    -l H3K4me3.bdgene-tss H3K4me3.bdgene-tts <span class="se">\</span>
    -o H3K4me3.bdgene.TSS_TTS.heatmap.pdf --plotmatrix 3x2 <span class="se">\</span>
    --centerlabel center -z cg chg chh --zMax <span class="m">0</span>.3 <span class="m">1</span> <span class="m">0</span>.01
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/plot-heatmap-4.png"><img alt="heatmap0" class="align-center" src="../_images/plot-heatmap-4.png" style="width: 400px; height: 500px;" /></a>
<p><strong>We also used DMtools in AZA-AML BS-Seq data</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ python3 dmPlotHeatmap.py -m GSM1329865.profile.cg GSM1329866.profile.cg GSM1329867.profile.cg GSM1329868.profile.cg <span class="se">\</span>
-l AZA-treated-1 AZA-treated-2 WT-1 WT-2 -o dnmt.heatmap.pdf <span class="se">\</span>
-sl TSS -el TTS --zMax <span class="m">0</span>.8 --colorMap Spectral_r --kmeans <span class="m">3</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/AZA-heatmap.png"><img alt="heatmap0" class="align-center" src="../_images/AZA-heatmap.png" style="width: 560px; height: 500px;" /></a>
<p><strong>AZA-AML vs AML DMR-related gene heatmap</strong></p>
<a class="reference internal image-reference" href="../_images/AZA-dmrgeneheatmap.png"><img alt="heatmap0" class="align-center" src="../_images/AZA-dmrgeneheatmap.png" style="width: 560px; height: 500px;" /></a>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">-f</span> <span class="pre">MRFILE</span> <span class="pre">[MRFILE</span> <span class="pre">...],</span> <span class="pre">--mrfile</span> <span class="pre">MRFILE</span> <span class="pre">[MRFILE</span> <span class="pre">...]</span></code></dt><dd><p>input methylevel files, wildtype.body.c*.txt</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-m</span> <span class="pre">MATRIXFILE</span> <span class="pre">[MATRIXFILE</span> <span class="pre">...],</span> <span class="pre">--matrixfile</span> <span class="pre">MATRIXFILE</span> <span class="pre">[MATRIXFILE</span> <span class="pre">...]</span></code></dt><dd><p>input methylevel matrix files, wildtype.GENE.cg.txt</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-l</span> <span class="pre">SAMPLESLABEL</span> <span class="pre">[SAMPLESLABEL</span> <span class="pre">...],</span> <span class="pre">--samplesLabel</span> <span class="pre">SAMPLESLABEL</span> <span class="pre">[SAMPLESLABEL</span> <span class="pre">...]</span></code></dt><dd><p>the label of the samples</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-z</span> <span class="pre">GROUPLABELS</span> <span class="pre">[GROUPLABELS</span> <span class="pre">...],</span> <span class="pre">--groupLabels</span> <span class="pre">GROUPLABELS</span> <span class="pre">[GROUPLABELS</span> <span class="pre">...]</span></code></dt><dd><p>Labels for the regions plotted in the heatmap. If more than one region is being plotted, a list of labels separated by spaces is
required. If a label itself contains a space, then quotes are needed. For example, --groupLabels label_1, &quot;label 2&quot;. should be
same length as --kmeans if kmeans valid</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-sl</span> <span class="pre">STARTLABEL,</span> <span class="pre">--startlabel</span> <span class="pre">STARTLABEL</span></code></dt><dd><p>the start label of the samples</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-el</span> <span class="pre">ENDLABEL,</span> <span class="pre">--endlabel</span> <span class="pre">ENDLABEL</span></code></dt><dd><p>the end label of the samples</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-pl</span> <span class="pre">CENTERLABEL,</span> <span class="pre">--centerlabel</span> <span class="pre">CENTERLABEL</span></code></dt><dd><p>the center label of the samples</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--plotmatrix</span> <span class="pre">PLOTMATRIX</span></code></dt><dd><p>1x1, default, row x col, order by columun, for exsample, 2x3 :file1 file2 file3file4 file5 file6</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--outFileName</span> <span class="pre">FILENAME,</span> <span class="pre">-o</span> <span class="pre">FILENAME</span></code></dt><dd><p>Output file name.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-c</span> <span class="pre">COLORMAP</span> <span class="pre">[COLORMAP</span> <span class="pre">...],</span> <span class="pre">--colorMap</span> <span class="pre">COLORMAP</span> <span class="pre">[COLORMAP</span> <span class="pre">...]</span></code></dt><dd><p>Color map to use for the heatmap. If more than one heatmap is being plotted the color of each heatmap can be enter individually
(e.g. <cite>--colorMap Reds Blues</cite>).The available options are: 'magma', 'inferno', 'plasma', 'viridis', 'cividis', 'twilight',
'twilight_shifted', 'turbo', 'Blues', 'BrBG', 'BuGn', 'BuPu', 'CMRmap', 'GnBu', 'Greens', 'Greys', 'OrRd', 'Oranges', 'PRGn',
'PiYG', 'PuBu', 'PuBuGn', 'PuOr', 'PuRd', 'Purples', 'RdBu', 'RdGy', 'RdPu', 'RdYlBu', 'RdYlGn', 'Reds', 'Spectral', 'Wistia',
'YlGn', 'YlGnBu', 'YlOrBr', 'YlOrRd', 'afmhot', 'autumn', 'binary', 'bone', 'brg', 'bwr', 'cool', 'coolwarm', 'copper',
'cubehelix', 'flag', 'gist_earth', 'gist_gray', 'gist_heat', 'gist_ncar', 'gist_rainbow', 'gist_stern', 'gist_yarg', 'gnuplot',
'gnuplot2', 'gray', 'hot', 'hsv', 'jet', 'nipy_spectral', 'ocean', 'pink', 'prism', 'rainbow', 'seismic', 'spring', 'summer',
'terrain', 'winter', 'Accent', 'Dark2', 'Paired', 'Pastel1', 'Pastel2', 'Set1', 'Set2', 'Set3', 'tab10', 'tab20', 'tab20b',
'tab20c', 'rocket', 'mako', 'icefire', 'vlag', 'flare', 'crest'</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--alpha</span> <span class="pre">ALPHA</span></code></dt><dd><p>The alpha channel (transparency) to use for the heatmaps. The default is 1.0 and values must be between 0 and 1.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--colorList</span> <span class="pre">COLORLIST</span> <span class="pre">[COLORLIST</span> <span class="pre">...]</span></code></dt><dd><p>List of colors to use to create a colormap. For example, if <cite>--colorList black,yellow,blue</cite> is set (colors separated by comas)
then a color map that starts with black, continues to yellow and finishes in blue is created. If this option is selected, it
overrides the --colorMap chosen. The list of valid color names can be seen here:
<a class="reference external" href="http://matplotlib.org/examples/color/named_colors.html">http://matplotlib.org/examples/color/named_colors.html</a> The number of transitions is defined by the --colorNumber option.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--colorNumber</span> <span class="pre">COLORNUMBER</span></code></dt><dd><p>--colorList is required for an effect. This controls the number of transitions from one color to the other. If --colorNumber is
the number of colors in --colorList then there will be no transitions between the colors.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--missingDataColor</span> <span class="pre">MISSINGDATACOLOR</span></code></dt><dd><p>If --missingDataAsZero was not set, such cases will be colored in white by default.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--sortRegions</span> <span class="pre">{descend,ascend,no}</span></code></dt><dd><p>Whether the heatmap should present the regions sorted. The default is to sort in descending order based on the mean value per
region.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--sortUsing</span> <span class="pre">{mean,median,max,min,sum}</span></code></dt><dd><p>Indicate which method should be used for sorting. For each row the method is computed.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--sortUsingSamples</span> <span class="pre">SORTUSINGSAMPLES</span> <span class="pre">[SORTUSINGSAMPLES</span> <span class="pre">...]</span></code></dt><dd><p>List of sample numbers (order as in matrix), which are used by --sortUsing for sorting. If no value is set, it uses all samples.
Example: --sortUsingSamples 1 3</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--linesAtTickMarks</span></code></dt><dd><p>Draw dashed lines from all tick marks through the heatmap. This is then similar to the dashed line draw at region bounds when
using a reference point and --sortUsing region_length</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--clusterUsingSamples</span> <span class="pre">CLUSTERUSINGSAMPLES</span> <span class="pre">[CLUSTERUSINGSAMPLES</span> <span class="pre">...]</span></code></dt><dd><p>List of sample numbers (order as in matrix), that are used for clustering by --kmeans or --hclust if not given, all samples are
taken into account for clustering. Example: --ClusterUsingSamples 1 3</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--kmeans</span> <span class="pre">KMEANS</span></code></dt><dd><p>Number of clusters to compute. When this option is set, the matrix is split into clusters using the k-means algorithm. Only works
for data that is not grouped, otherwise only the first group will be clustered.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--hclust</span> <span class="pre">HCLUST</span></code></dt><dd><p>Number of clusters to compute. When this option is set, then the matrix is split into clusters using the hierarchical clustering
algorithm, using &quot;ward linkage&quot;. Only works for data that is not grouped, otherwise only the first group will be clustered.
--hclust could be very slow if you have &gt;1000 regions. In those cases, you might prefer --kmeans or if more clustering methods are
required you can save the underlying matrix and run the clustering using other software. The plotting of the clustering may fail
with an error if a cluster has very few members compared to the total number of regions.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-s</span> <span class="pre">SCALE</span> <span class="pre">[SCALE</span> <span class="pre">...],</span> <span class="pre">--scale</span> <span class="pre">SCALE</span> <span class="pre">[SCALE</span> <span class="pre">...]</span></code></dt><dd><p>Maximum value for the Y-axis. Multiple values, separated by spaces can be set for each profile. If the number of yMin values is
smaller thanthe number of plots, the values are recycled.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-t</span> <span class="pre">TITLE</span> <span class="pre">[TITLE</span> <span class="pre">...],</span> <span class="pre">--title</span> <span class="pre">TITLE</span> <span class="pre">[TITLE</span> <span class="pre">...]</span></code></dt><dd><p>Title of the plot, to be printed on top of the generated image. Leave blank for no title.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--zMin</span> <span class="pre">ZMIN</span> <span class="pre">[ZMIN</span> <span class="pre">...]</span></code></dt><dd><p>Values to anchor the colormap</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--zMax</span> <span class="pre">ZMAX</span> <span class="pre">[ZMAX</span> <span class="pre">...]</span></code></dt><dd><p>Values to anchor the colormap, Maximum value for the heatmap.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-ft</span> <span class="pre">IMAGE_FORMAT,</span> <span class="pre">--image_format</span> <span class="pre">IMAGE_FORMAT</span></code></dt><dd><p>The file format, e.g. 'png', 'pdf', 'svg', ... The behavior when this is unset is documented under fname.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--perGroup</span></code></dt><dd><p>The default is to plot all groups of regions by sample. Using this option instead plots all samples by group of regions. Note that
this is only useful if you have multiple groups of regions. by sample rather than group.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--dpi</span> <span class="pre">DPI</span></code></dt><dd><p>Set the DPI to save the figure. default: 100</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--figsize</span> <span class="pre">FIGSIZE</span></code></dt><dd><p>Set the figure size to save the figure. [with]x[height], default: 1.5x11</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--boxAroundHeatmaps</span> <span class="pre">BOXAROUNDHEATMAPS</span></code></dt><dd><p>By default black boxes are plot around heatmaps. This can be turned off by setting --boxAroundHeatmaps no</p>
</dd>
</dl>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="dmplotcor.html" class="btn btn-neutral float-left" title="dmPlotCor" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="dmplotprofile.html" class="btn btn-neutral float-right" title="dmPlotProfile" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>