<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmPlotCor &mdash; dmtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            dmtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="dmtools.html">DMtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">DiffMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pydmtools.html">pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">dmtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">dmPlotCor</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/dmplotcor.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmplotcor">
<h1>dmPlotCor<a class="headerlink" href="#dmplotcor" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage" id="id2">Usage</a></p></li>
<li><p><a class="reference internal" href="#parameters" id="id3">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>In DNA methylation data analysis, calculating the correlation between biological replicates
and different samples is essential. Biological replicates refer to data obtained by sequencing
or performing the same experiment multiple times on the same sample.
By calculating biological replicates, we can evaluate the reproducibility and stability of the data,
and determine whether changes in methylation levels are due to true biological differences or
technical noise.</p>
<p>On the other hand, the correlation between different samples can help us evaluate their similarity
or differences. By calculating the correlation between samples, we can identify which samples
have similar methylation patterns and which samples exhibit differences.
This helps us identify critical differential methylation sites and determine the similarity
and differences in methylation patterns among different samples,
leading to more accurate conclusions.</p>
<p>Therefore, in DNA methylation data analysis, calculating biological replicates and the correlation
between different samples is a crucial step that helps us evaluate the reliability and
consistency of the data.</p>
<p>We can use <a class="reference internal" href="chromstats.html"><span class="doc">dmtools chromstats</span></a> to calculate the DNA methylation level in genomic regions
of different lengths and calculate the correlation between different samples based on the results.</p>
</section>
<section id="usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage</a><a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p>DMtools provides two visualization options, heatmap and scatterplot, to calculate
the correlation between DNA methylation data using two methods: Spearman and Pearson.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools chromstats -i sample1.zm0.dm -o sample1.chrmeth.txt
$ dmtools chromstats -i sample2.zm0.dm -o sample2.chrmeth.txt
<span class="c1"># same process for other samples</span>

$ python3 dmPlotCor.py --plotFile bs.correlation.pdf -m spearman -p heatmap <span class="se">\</span>
    --plotNumbers -f sample1.chrmeth.txt sample2.chrmeth.txt sample3.chrmeth.txt <span class="se">\</span>
    sample4.chrmeth.txt -c CG -l s1_rep1 s1_rep2 s2_rep1 s2_rep2
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/cor.heatmap.png"><img alt="heatmap" class="align-center" src="../_images/cor.heatmap.png" style="width: 360px; height: 320px;" /></a>
<p>scatterplot for correlation visulization</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ python3 dmPlotCor.py --plotFile bs.correlation.pdf -m spearman -p scatterplot <span class="se">\</span>
    --plotNumbers -f sample1.chrmeth.txt sample2.chrmeth.txt sample3.chrmeth.txt <span class="se">\</span>
    sample4.chrmeth.txt -c CG -l AZA_rep1 AZA_rep2 AML_rep1 AML_rep2 <span class="se">\</span>
    --xRange <span class="m">0</span> <span class="m">1</span> --yRange <span class="m">0</span> <span class="m">1</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/scatterplot.png"><img alt="scatterplot" class="align-center" src="../_images/scatterplot.png" style="width: 380px; height: 380px;" /></a>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p><strong>options:</strong></p>
<p><code class="docutils literal notranslate"><span class="pre">-f</span> <span class="pre">BINFILES</span> <span class="pre">[BINFILES</span> <span class="pre">...],</span> <span class="pre">--binFiles</span> <span class="pre">BINFILES</span> <span class="pre">[BINFILES</span> <span class="pre">...]</span></code></p>
<blockquote>
<div><p>the list of input DNA methylation files to merge and plot, must provide -f or -i</p>
</div></blockquote>
<p><code class="docutils literal notranslate"><span class="pre">--corMatrix</span> <span class="pre">FILE,</span> <span class="pre">-i</span> <span class="pre">FILE</span></code></p>
<blockquote>
<div><p>Matrix of DNA methylation values, must provide -f or -i</p>
</div></blockquote>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">--corMethod</span> <span class="pre">{spearman,pearson},</span> <span class="pre">-m</span> <span class="pre">{spearman,pearson}</span></code></dt><dd><p>Correlation method.</p>
</dd>
</dl>
<p><code class="docutils literal notranslate"><span class="pre">--whatToPlot</span> <span class="pre">{heatmap,scatterplot},</span> <span class="pre">-p</span> <span class="pre">{heatmap,scatterplot}</span></code></p>
<blockquote>
<div><p>Choose between a heatmap or pairwise scatter plots</p>
</div></blockquote>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">--context</span> <span class="pre">{All,C,CG,CHG,CHH},</span> <span class="pre">-c</span> <span class="pre">{All,C,CG,CHG,CHH}</span></code></dt><dd><p>DNA methylation context for calculate, All equal to CG+CHG+CHH, C means average of CG/CHG/CHH</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--plotFile</span> <span class="pre">FILE,</span> <span class="pre">-o</span> <span class="pre">FILE</span></code></dt><dd><p>File to save the heatmap to. The file extension determines the format, so heatmap.pdf will save the heatmap in PDF format. The
available formats are: .png, .eps, .pdf and .svg.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--skipZeros</span></code></dt><dd><p>By setting this option, genomic regions that have zero or missing (nan) values in all samples are excluded.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--skipOnes</span></code></dt><dd><p>By setting this option, genomic regions that have DNA methylation 100 percentage values in all samples are excluded.</p>
</dd>
</dl>
<p><code class="docutils literal notranslate"><span class="pre">--labels</span> <span class="pre">sample1</span> <span class="pre">sample2</span> <span class="pre">[sample1</span> <span class="pre">sample2</span> <span class="pre">...],</span> <span class="pre">-l</span> <span class="pre">sample1</span> <span class="pre">sample2</span> <span class="pre">[sample1</span> <span class="pre">sample2</span> <span class="pre">...]</span></code></p>
<blockquote>
<div><p>User defined labels instead of default labels from file names. Multiple labels have to be separated by spaces, e.g. --labels
sample1 sample2 sample3</p>
</div></blockquote>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">--plotTitle</span> <span class="pre">PLOTTITLE,</span> <span class="pre">-T</span> <span class="pre">PLOTTITLE</span></code></dt><dd><p>Title of the plot, to be printed on top of the generated image. Leave blank for no title. (Default: )</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--plotFileFormat</span> <span class="pre">FILETYPE,</span> <span class="pre">-pf</span> <span class="pre">FILETYPE</span></code></dt><dd><p>Image format type. If given, this option overrides the image format based on the plotFile ending. The available options are: png,
eps, pdf and svg.</p>
</dd>
</dl>
<p><strong>Output optional options:</strong></p>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">--outFileCorMatrix</span> <span class="pre">FILE</span></code></dt><dd><p>Save matrix with pairwise correlation values to a tab-separated file.</p>
</dd>
</dl>
<p><strong>Heatmap options:</strong></p>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">--plotHeight</span> <span class="pre">PLOTHEIGHT</span></code></dt><dd><p>Plot height in cm. (Default: 9.5)</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--plotWidth</span> <span class="pre">PLOTWIDTH</span></code></dt><dd><p>Plot width in cm. The minimum value is 1 cm. (Default: 11)</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--zMin</span> <span class="pre">ZMIN,</span> <span class="pre">-min</span> <span class="pre">ZMIN</span></code></dt><dd><p>Minimum value for the heatmap intensities. If not specified, the value is set automatically</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--zMax</span> <span class="pre">ZMAX,</span> <span class="pre">-max</span> <span class="pre">ZMAX</span></code></dt><dd><p>Maximum value for the heatmap intensities.If not specified, the value is set automatically</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--colorMap</span></code></dt><dd><p>Color map to use for the heatmap. Available values can be seen here: <a class="reference external" href="http://matplotlib.org/examples/color/colormaps_reference.html">http://matplotlib.org/examples/color/colormaps_reference.html</a></p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--plotNumbers</span></code></dt><dd><p>If set, then the correlation number is plotted on top of the heatmap. This option is only valid when plotting a heatmap.</p>
</dd>
</dl>
<p><strong>Scatter plot options:</strong></p>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">--xRange</span> <span class="pre">XRANGE</span> <span class="pre">XRANGE</span></code></dt><dd><p>The X axis range. The default scales these such that the full range of dots is displayed.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">--yRange</span> <span class="pre">YRANGE</span> <span class="pre">YRANGE</span></code></dt><dd><p>The Y axis range. The default scales these such that the full range of dots is displayed.</p>
</dd>
</dl>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>