<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmtools profile &mdash; DMtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            DMtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Getting started with dmtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmtools.html">dmtools: command line tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmplot.html">dmplot: plotting from the command line</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dmdiff.html">dmdiff: differential dna methylation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pydmtools.html">Python API: pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">DMtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">dmtools profile</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/profile.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmtools-profile">
<h1>dmtools profile<a class="headerlink" href="#dmtools-profile" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage" id="id2">Usage</a></p>
<ul>
<li><p><a class="reference internal" href="#with-gtf-gff-file" id="id3">With gtf/gff file</a></p></li>
<li><p><a class="reference internal" href="#or-with-bed-file" id="id4">Or with bed file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#parameters" id="id5">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>DMtools can also calculate the DNA methylation profiles around genes,
introns, exons, intergenic regions and TEs.</p>
<p>Calculating the distribution pattern of DNA methylation levels across genes and their
upstream and downstream regions is crucial for understanding the role of gene regulation and epigenetics.
Changes in the methylation levels of genes and promoter regions can affect gene expression and function.
Previous studies have shown that high DNA methylation in promoter regions suppresses
gene expression, while the opposite is true for gene regions.
The distribution pattern of methylation levels in different genes and their upstream and downstream
regions across different samples may be associated with specific biological processes and diseases.
Therefore, by calculating and comparing the distribution patterns of DNA methylation levels in
different samples and different genes' upstream and downstream regions within the same sample,
we can better understand the role of DNA methylation in gene regulation and epigenetics.</p>
</section>
<section id="usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage</a><a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p>Calculate DNA methylation profile matrix and average matrix across gene body, upstream and downstream:</p>
<section id="with-gtf-gff-file">
<h3><a class="toc-backref" href="#id3" role="doc-backlink">With gtf/gff file</a><a class="headerlink" href="#with-gtf-gff-file" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools profile -i sample1.methratio.dm --gtf gene.gtf -o gene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">0</span>
</pre></div>
</div>
</section>
<section id="or-with-bed-file">
<h3><a class="toc-backref" href="#id4" role="doc-backlink">Or with bed file</a><a class="headerlink" href="#or-with-bed-file" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools profile -i sample1.methratio.dm --bed gene.bed -o gene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">0</span>
</pre></div>
</div>
<p>Please see 'dmtools profile' for more details.</p>
</section>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">-i</span></code> input DM file</p>
<p><code class="docutils literal notranslate"><span class="pre">--bed</span></code> bed file for view, format: chrom start end [strand].</p>
<p><code class="docutils literal notranslate"><span class="pre">--gtf</span></code> gtf file for view, format: chrom * * start end * strand * xx geneid.</p>
<p><code class="docutils literal notranslate"><span class="pre">--gff</span></code> gff file for view, format: chrom * * start end * strand * xx=geneid.</p>
<p><code class="docutils literal notranslate"><span class="pre">-o</span></code> output file [stdout]</p>
<p><code class="docutils literal notranslate"><span class="pre">-p</span></code> [int] threads</p>
<p><code class="docutils literal notranslate"><span class="pre">--regionextend</span></code> region extend for upstream and downstram, [2000]</p>
<p><code class="docutils literal notranslate"><span class="pre">--profilestep</span></code> [double] step mean bin size for chromosome region, default: 0.02 (2%)</p>
<p><code class="docutils literal notranslate"><span class="pre">--profilemovestep</span></code> [double] step move, default: profilestep/2, if no overlap, please define same as --profilestep</p>
<p><code class="docutils literal notranslate"><span class="pre">--profilemode</span></code> calculate the methylation matrix mode of every region or gene. 0 for gene and flanks mode, 1 for tss and flanks, 2 for tts and flanks, 3 for region center and flanks. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">--bodyX</span></code> [double] the gene body bin size is N multiple of the bin size of the upstream and downstream extension region. [1]</p>
<p><code class="docutils literal notranslate"><span class="pre">--matrixX</span></code> [int] the bin size is N times of the profile bin size, so the bin size should be N*profilestep [5], please note N*profilestep must &lt; 1 and N must &gt;= 1, used for calculation per gene.</p>
<p><code class="docutils literal notranslate"><span class="pre">--strand</span></code> [0/1/2] strand for calculate, 0 represent '+' positive strand, 1 '-' negative strand, 2 '.' all information, [2]</p>
<p><code class="docutils literal notranslate"><span class="pre">--context</span></code> [0/1/2/3] context for calculate, 0 represent 'C/ALL' context, 1 'CG' context, 2 'CHG' context, 3 'CHH' context. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">--print2one</span></code> [int] print all the matrix results of C/CG/CHG/CHH context methylation to same file. 0 for no, 1 for yes. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">-h|--help</span></code></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>