<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmtools regionstats &mdash; DMtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="dmtools bodystats" href="bodystats.html" />
    <link rel="prev" title="dmtools stats" href="stats.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            DMtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Getting started with dmtools</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../dmtools.html">dmtools: command line tools</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bam2dm.html">Calculate DNA meth from bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="ebsrate.html">Estimate bs conversion rate</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="dmtools-calme.html">Calculate DNA meth with dm</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="stats.html">Calculate DNA methylation for dm file</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Calculate DNA methylation for regions</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l4"><a class="reference internal" href="#usage-and-output">Usage and output</a></li>
<li class="toctree-l4"><a class="reference internal" href="#parameters">Parameters</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="bodystats.html">Calculate DNA methylation for gene body</a></li>
<li class="toctree-l3"><a class="reference internal" href="profile.html">Calculate methylation profile across genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="chromstats.html">Calculate DNA methylation acorss chromosome</a></li>
<li class="toctree-l3"><a class="reference internal" href="chrmeth.html">Calculate DNA methylation for chromosome</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-merge.html">Merge or overlap dm files</a></li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-manu.html">Manipulate dm format</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../dmplot.html">dmplot: plotting from the command line</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">dmdiff: differential dna methylation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pydmtools.html">Python API: pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">DMtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../dmtools.html">Getting started</a></li>
          <li class="breadcrumb-item"><a href="dmtools-calme.html">DMtools main modules</a></li>
      <li class="breadcrumb-item active">dmtools regionstats</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/regionstats.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmtools-regionstats">
<h1>dmtools regionstats<a class="headerlink" href="#dmtools-regionstats" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage-and-output" id="id2">Usage and output</a></p>
<ul>
<li><p><a class="reference internal" href="#with-gtf-gff-file" id="id3">with gtf/gff file</a></p></li>
<li><p><a class="reference internal" href="#only-print-methylation-level-without-coverage" id="id4">only print methylation level without coverage</a></p></li>
<li><p><a class="reference internal" href="#or-with-bed-file" id="id5">Or with bed file</a></p></li>
<li><p><a class="reference internal" href="#or-just-calculate-dna-methylation-level-of-provide-regions" id="id6">Or just calculate DNA methylation level of provide regions:</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#parameters" id="id7">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>DMtools provides an <code class="docutils literal notranslate"><span class="pre">regionstats</span></code> command that can be used to calculate the DNA methylation level
within a user-specified region of interest.
This functionality is highly useful for researchers who may be interested in the methylation level
of specific genes, transposable elements (TEs), or other regions of interest.
By using the dmtools regionstats command, researchers can quickly and easily calculate the
methylation level of each region of interest, and perform further analysis and comparison.
This is crucial for uncovering the role of DNA methylation in various biological processes
and its association with diseases.</p>
<p>To calculate the methylation density level in a given genomic region,
only cytosines with coverage greater than the preset threshold are used.
The DNA methylation level in a genomic region is defined as the total number of sequenced Cs over the
total number of sequenced Cs and Ts at all cytosine positions across the region,
and the equation is as follows:</p>
<div class="math">
<p><img src="../_images/math/5df321b8f1e45c7229dcb040c5c7a54e1ddf7be6.png" alt="ML(x)_{method} = \begin{cases}
\dfrac{\sum_{i=1}^{N} C}{\sum_{i=1}^{N} (C + T)}, &amp; \text{if method = weighted} \\
\dfrac{\sum_{i=1}^{N} ML(x_{i})}{N}, &amp; \text{if method = mean} \\
\end{cases}"/></p>
</div><p>where N is the total number of cytosine sites whose coverage is more than the predefined threshold in the genomic region.</p>
<p>Calculate DNA methylation level of chromosome region, genes, or TEs:</p>
</section>
<section id="usage-and-output">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage and output</a><a class="headerlink" href="#usage-and-output" title="Permalink to this heading"></a></h2>
<section id="with-gtf-gff-file">
<h3><a class="toc-backref" href="#id3" role="doc-backlink">with gtf/gff file</a><a class="headerlink" href="#with-gtf-gff-file" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools regionstats -i sample1.methratio.dm --gtf gene.gtf -o gene.meth.txt --printcoverage <span class="m">1</span>
  <span class="c1">## chromosome pos strand meth coverage geneid</span>
  <span class="c1">#Chr1     4396348 -       6       567     AT1G12920</span>
  <span class="c1">#Chr1     4396348 -       12      1552    AT1G12920</span>
  <span class="c1">#Chr1     4398375 -       114     3381    AT1G12930</span>
</pre></div>
</div>
</section>
<section id="only-print-methylation-level-without-coverage">
<h3><a class="toc-backref" href="#id4" role="doc-backlink">only print methylation level without coverage</a><a class="headerlink" href="#only-print-methylation-level-without-coverage" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools regionstats -i sample1.methratio.dm --gtf gene.gtf -o gene.meth.txt --printcoverage <span class="m">0</span>
  <span class="c1">## chromosome pos methy-level geneid</span>
  <span class="c1">#Chr1     1618602 -       0.009665        AT1G05490</span>
  <span class="c1">#Chr1     1618602 -       0.014290        AT1G05490</span>
  <span class="c1">#Chr1     1624955 +       0.048446        AT1G05500</span>
  <span class="c1">#Chr1     1624955 +       0.213080        AT1G05500</span>
</pre></div>
</div>
</section>
<section id="or-with-bed-file">
<h3><a class="toc-backref" href="#id5" role="doc-backlink">Or with bed file</a><a class="headerlink" href="#or-with-bed-file" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools regionstats -i sample1.methratio.dm --bed gene.bed -o gene.meth.txt
</pre></div>
</div>
</section>
<section id="or-just-calculate-dna-methylation-level-of-provide-regions">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Or just calculate DNA methylation level of provide regions:</a><a class="headerlink" href="#or-just-calculate-dna-methylation-level-of-provide-regions" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools regionstats -i sample1.methratio.dm -r chr1:1-2900<span class="p">;</span>chr2:1-200,+ <span class="se">\</span>
  -o gene.meth.txt
</pre></div>
</div>
<p>Please see 'dmtools regionstats' for more details.</p>
</section>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id7" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">-i</span></code> input DM file</p>
<p><code class="docutils literal notranslate"><span class="pre">--bed</span></code> bed file for view, format: chrom start end [strand].</p>
<p><code class="docutils literal notranslate"><span class="pre">--gtf</span></code> gtf file for view, format: chrom * * start end * strand * xx geneid.</p>
<p><code class="docutils literal notranslate"><span class="pre">--gff</span></code> gff file for view, format: chrom * * start end * strand * xx=geneid.</p>
<p><code class="docutils literal notranslate"><span class="pre">-o</span></code> output prefix [stdout]</p>
<p><code class="docutils literal notranslate"><span class="pre">-r</span></code> region for view, can be seperated by space. chr1:1-2900 chr2:1-200,+</p>
<p><code class="docutils literal notranslate"><span class="pre">--method</span></code> weighted/ mean</p>
<p><code class="docutils literal notranslate"><span class="pre">--strand</span></code> [0/1/2/3] strand for show, 0 represent '+' positive strand, 1 '-' negative strand, 2 '.' all information, 3 calculate and print strand meth level seperately</p>
<p><code class="docutils literal notranslate"><span class="pre">--context</span></code> [0/1/2/3/4] context for show, 0 represent 'C/ALL' context, 1 'CG' context, 2 'CHG' context, 3 'CHH' context, 4 calculate and print strand meth level seperately</p>
<p><code class="docutils literal notranslate"><span class="pre">--printcoverage</span></code> [0/1] print countC and coverage instead of methratio. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">--print2one</span></code> [int] print all the countC and coverage results of C/CG/CHG/CHH context methylation to same file, only valid when --printcoverage 1. 0 for no, 1 for yes. [0]</p>
<p><code class="docutils literal notranslate"><span class="pre">-h|--help</span></code></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="stats.html" class="btn btn-neutral float-left" title="dmtools stats" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="bodystats.html" class="btn btn-neutral float-right" title="dmtools bodystats" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>