<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>dmtools stats &mdash; DMtools 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="dmtools regionstats" href="regionstats.html" />
    <link rel="prev" title="DMtools main modules" href="dmtools-calme.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            DMtools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">methylation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Getting started with dmtools</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../dmtools.html">dmtools: command line tools</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="align.html">Process raw fastq data</a></li>
<li class="toctree-l2"><a class="reference internal" href="bam2dm.html">Calculate DNA meth from bam</a></li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-format.html">Format convertion</a></li>
<li class="toctree-l2"><a class="reference internal" href="ebsrate.html">Estimate bs conversion rate</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="dmtools-calme.html">Calculate DNA meth with dm</a><ul class="current">
<li class="toctree-l3 current"><a class="current reference internal" href="#">Calculate DNA methylation for dm file</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l4"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l4"><a class="reference internal" href="#output">Output</a></li>
<li class="toctree-l4"><a class="reference internal" href="#parameters">Parameters</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="regionstats.html">Calculate DNA methylation for regions</a></li>
<li class="toctree-l3"><a class="reference internal" href="bodystats.html">Calculate DNA methylation for gene body</a></li>
<li class="toctree-l3"><a class="reference internal" href="profile.html">Calculate methylation profile across genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="chromstats.html">Calculate DNA methylation acorss chromosome</a></li>
<li class="toctree-l3"><a class="reference internal" href="chrmeth.html">Calculate DNA methylation for chromosome</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-merge.html">Merge or overlap dm files</a></li>
<li class="toctree-l2"><a class="reference internal" href="dmtools-manu.html">Manipulate dm format</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../dmplot.html">dmplot: plotting from the command line</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">dmdiff: differential dna methylation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pydmtools.html">Python API: pydmtools</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">DMtools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../dmtools.html">Getting started</a></li>
          <li class="breadcrumb-item"><a href="dmtools-calme.html">DMtools main modules</a></li>
      <li class="breadcrumb-item active">dmtools stats</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/stats.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="dmtools-stats">
<h1>dmtools stats<a class="headerlink" href="#dmtools-stats" title="Permalink to this heading"></a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#description" id="id1">Description</a></p></li>
<li><p><a class="reference internal" href="#usage" id="id2">Usage</a></p>
<ul>
<li><p><a class="reference internal" href="#calculate-dna-methylation-data-coverage-and-dna-methylation-level-category" id="id3">Calculate DNA methylation data coverage and DNA methylation level category</a></p></li>
<li><p><a class="reference internal" href="#count-number-of-gc-of-genome" id="id4">Count number of GC of genome</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id5">Output</a></p></li>
<li><p><a class="reference internal" href="#parameters" id="id6">Parameters</a></p></li>
</ul>
</nav>
<section id="description">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Description</a><a class="headerlink" href="#description" title="Permalink to this heading"></a></h2>
<p>In DNA methylation analysis, coverage refers to the number of sequence reads in which information about each cytosine site can be detected. Higher coverage ensures data reliability and accuracy. Therefore, it is important to calculate the number and proportion of cytosine sites that meet a specific coverage threshold.
By calculating the number and proportion of cytosine sites with coverage greater than or equal to a specific threshold, we can assess the quality and reliability of DNA methylation data. A coverage threshold is usually chosen to ensure data accuracy and reliability. Additionally, coverage thresholds may vary for different samples, and calculating the number and proportion of cytosine sites at different coverage levels can better compare differences and similarities between samples.
In summary, calculating the number and proportion of cytosine sites that meet a specific coverage threshold is a crucial step in DNA methylation data analysis, as it helps us evaluate data quality and reliability.</p>
<p>Dividing DNA methylation levels into different intervals can help us better understand the distribution of DNA methylation in different regions. By calculating the number of DNA methylation sites in each interval, we can more accurately assess the degree of methylation at different methylation levels. This is essential for understanding the role of DNA methylation in gene regulation and epigenetics.
For example, we can divide DNA methylation levels into different intervals, such as 0-25%, 25-50%, 50-75%, 75-100%, and calculate the number and proportion of DNA methylation sites in each interval. This can help us gain a more comprehensive understanding of the role of DNA methylation in gene expression and functional regulation. In addition, the distribution of DNA methylation in different samples may differ. By calculating the number and proportion of DNA methylation sites in different intervals among different samples, we can compare the differences and similarities between different samples.
In summary, dividing DNA methylation levels into different intervals and calculating the number of DNA methylation sites in each interval is a crucial step in DNA methylation data analysis, which can evaluate the differences and similarities between different samples.</p>
<p>The DMtools <code class="docutils literal notranslate"><span class="pre">stats</span></code> command can calculate the number and proportion of cytosine sites that meet different coverage thresholds, as well as the number of cytosine sites at different DNA methylation levels.</p>
</section>
<section id="usage">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Usage</a><a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<section id="calculate-dna-methylation-data-coverage-and-dna-methylation-level-category">
<h3><a class="toc-backref" href="#id3" role="doc-backlink">Calculate DNA methylation data coverage and DNA methylation level category</a><a class="headerlink" href="#calculate-dna-methylation-data-coverage-and-dna-methylation-level-category" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ dmtools stats -i sample1.methratio.dm -o chromosome.cover --tc <span class="m">1200559022</span>
</pre></div>
</div>
<p><cite>--tc</cite> is the total number of C and G in the genome, we can obtained by 'python count_cg.py genome.fa' in DMtools dir.</p>
</section>
<section id="count-number-of-gc-of-genome">
<h3><a class="toc-backref" href="#id4" role="doc-backlink">Count number of GC of genome</a><a class="headerlink" href="#count-number-of-gc-of-genome" title="Permalink to this heading"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ python count_cg.py hg38.chr.fa
<span class="c1">#599043897 C, 601515125 G and 1200559022 CG in genome</span>
</pre></div>
</div>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<ol class="arabic simple">
<li><p>Calculate the number of cytosine sites with coverage greater than or equal to N, as well as the proportion of these sites among all cytosine sites.</p></li>
</ol>
<blockquote>
<div><p><cite>chromosome.cover</cite></p>
</div></blockquote>
<p>For example, in this test sample, the number of sites with DNA methylation coverage greater than 1 is 803251004, which accounts for 0.67 of all genomic cytosine sites.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#cover   1       2       3       4       5       6       7       8       9       10      11      12      13      14      15      16</span>
<span class="c1">#count   803251004       803251004       803251004       803251004       803251004       708649278       616748604       530217962       450748375       379329620       316287005       261534457       214678989       175127002       142130490       114926346</span>
<span class="c1">#percent 0.67    0.67    0.67    0.67    0.67    0.59    0.51    0.44    0.38    0.32    0.26    0.22    0.18    0.15    0.12    0.10</span>
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Calculate the number of cytosine sites at different methylation levels. <code class="docutils literal notranslate"><span class="pre">-s</span></code> parameter change the size number.</p></li>
</ol>
<p><cite>chromosome.cover.stats</cite></p>
<p>For example, in this test sample, the number of sites with DNA methylation levels between [0-0.1) is 760990560.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">#catary  0       0.10    0.20    0.30    0.40    0.50    0.60    0.70    0.80    0.90</span>
<span class="c1">#percent 760990560       19059281        4978777 6203698 5038056 2986205 1880708 1318675 561500  233544</span>
</pre></div>
</div>
</section>
<section id="parameters">
<h2><a class="toc-backref" href="#id6" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">-i</span></code> input DM file</p>
<p><code class="docutils literal notranslate"><span class="pre">-o</span></code> output file [stdout]</p>
<p><code class="docutils literal notranslate"><span class="pre">--tc</span></code> total number of cytosine and guanine in genome, we will use the number of site in dm file if you not provide --tc</p>
<p><code class="docutils literal notranslate"><span class="pre">-r</span></code> region for calculate stats, can be seperated by space. chr1:1-2900 chr2:1-200</p>
<p><code class="docutils literal notranslate"><span class="pre">--bed</span></code> bed file for calculate stats, format: chrom start end (strand).</p>
<p><code class="docutils literal notranslate"><span class="pre">--strand</span></code> [0/1/2] strand for show, 0 represent '+' positive strand, 1 '-' negative strand, 2 '.' all information</p>
<p><code class="docutils literal notranslate"><span class="pre">--context</span></code> [0/1/2/3] context for show, 0 represent 'C/ALL' context, 1 'CG' context, 2 'CHG' context, 3 'CHH' context.</p>
<p><code class="docutils literal notranslate"><span class="pre">-s</span></code> size of bin for count cytosine number.</p>
<p><code class="docutils literal notranslate"><span class="pre">--mincover</span></code> &gt;= minumum coverage show, default: 0</p>
<p><code class="docutils literal notranslate"><span class="pre">--maxcover</span></code> &lt;= maximum coverage show, default: 10000</p>
<p><code class="docutils literal notranslate"><span class="pre">-h|--help</span></code></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/dmtools">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="dmtools-calme.html" class="btn btn-neutral float-left" title="DMtools main modules" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="regionstats.html" class="btn btn-neutral float-right" title="dmtools regionstats" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>